keras_retinanet/bin/evaluate_coco.py

```diff
@@ -41,20 +41,22 @@ def get_session():
     return tf.Session(config=config)
 
 
-def parse_args():
+def parse_args(args):
     parser = argparse.ArgumentParser(description='Simple training script for COCO object detection.')
     parser.add_argument('model', help='Path to RetinaNet model.')
     parser.add_argument('coco_path', help='Path to COCO directory (ie. /tmp/COCO).')
     parser.add_argument('--gpu', help='Id of the GPU to use (as reported by nvidia-smi).')
     parser.add_argument('--set', help='Name of the set file to evaluate (defaults to val2017).', default='val2017')
     parser.add_argument('--score-threshold', help='Threshold on score to filter detections with (defaults to 0.05).', default=0.05, type=float)
 
-    return parser.parse_args()
+    return parser.parse_args(args)
 
 
-def main():
+def main(args=None):
     # parse arguments
-    args = parse_args()
+    if args is None:
+        args = sys.argv[1:]
```

Comment:
Did it this way (default to `None`, check in body) because otherwise the default value of `args` would be determined at parse time. If `args` defaults to `sys.argv[1:]` directly, updating `sys.argv` and then calling `main` would not change the default value (because it was determined at parse time).